research/long-context-llama/H2O/README.md

```diff
@@ -0,0 +1,11 @@
+## Run Llama with H2O for long context inference
+
+The following example runs inference of Llama-2-7b on XSUM summarization tasks. We're using `--enable_h2o_generation` to enable H2O algorithm that only keeps heavy-hitter and the local KV pairs. Use `--num_heavy_hitter_tokens` to decide the number of heavy-hitter KV pairs and `--num_window_length `for the KV cache size. The number of local KV pairs equals num_window_length - num_heavy_hitter_tokens. Also, use --enable_position_rolling to enable position rolling in the KV cache size that assign the positions in the KV cache instead of the ones in original sequences. Enabling postional rolling is important when sequence length exceeds the pretrained context windows, e.g., 4K in Llama-2.
+
+```
```

Comment:
running the code it runs into this error seems one arg is missing, 

```
 File "/data/users/hamidnazeri/fbsource/llama-recipe-new/llama-recipes/research/long-context-llama/H2O/utils/llama.py", line 290, in enable_h2ocache_forward
    causal_mask = self._update_causal_mask(attention_mask, inputs_embeds, cache_position)
TypeError: LlamaModel._update_causal_mask() takes 3 positional arguments but 4 were given
```

Reply:
can we also start adding numbers/ visuals how it looks/ what it improves/?

Reply:
can you please try moving the folder under `recipes/experimental/long-context/H2O`.